pintc/src/predicate/analyse/type_check/check_exprs.rs

```diff
@@ -1388,11 +1468,25 @@ impl Contract {
             return Inference::Dependant(index_expr_key);
         }
 
-        let ary_ty = array_expr_key.get_ty(self);
-        if ary_ty.is_unknown() {
+        let ty = array_expr_key.get_ty(self);
+        if ty.is_unknown() {
             return Inference::Dependant(array_expr_key);
         }
 
+        let is_storage_access = array_expr_key.is_storage_access(self);
```

Comment:
All of the boilerplate that this adds is unfortunate.  I feel like one day we should refactor it out to be a recursive call or something, if possible.

Reply:
Created an issue.